catalog/spatial/wfs/1.1.0/spatial-wfs-v1_1_0-converter/src/test/java/org/codice/ddf/spatial/ogc/wfs/v1_1_0/catalog/converter/impl/GenericFeatureConverterTest.java

```diff
@@ -0,0 +1,256 @@
+    assertThat(wfc.getFeatureMembers(), hasSize(1));
+    Metacard mc = wfc.getFeatureMembers().get(0);
+    assertThat(mc.getId(), is("video_data_set.1"));
+    assertThat(mc.getLocation(), nullValue());
+  }
+
+  @Test
+  public void testUnmarshalFeatureCollectionXmlToObject() {
+    XStream xstream = new XStream(new WstxDriver());
+    FeatureCollectionConverterWfs11 fcConverter = new FeatureCollectionConverterWfs11();
+    Map<String, FeatureConverter> fcMap = new HashMap<>();
+
+    GenericFeatureConverter converter = new GenericFeatureConverter();
+
+    fcMap.put("video_data_set", converter);
+    fcConverter.setFeatureConverterMap(fcMap);
+
+    xstream.registerConverter(fcConverter);
+
+    converter.setMetacardType(buildMetacardType());
+    xstream.registerConverter(converter);
+    xstream.registerConverter(new GmlGeometryConverter());
+    xstream.alias("FeatureCollection", WfsFeatureCollection.class);
+    InputStream is =
+        GenericFeatureConverterTest.class.getResourceAsStream("/video_data_set_collection.xml");
+
+    WfsFeatureCollection wfc = (WfsFeatureCollection) xstream.fromXML(is);
+    assertThat(wfc.getFeatureMembers(), hasSize(4));
+    Metacard mc = wfc.getFeatureMembers().get(0);
+    assertThat(mc.getId(), is("video_data_set.1"));
+  }
+
+  @Test
+  public void testUnmarshalFeatureCollectionFeatureMembersXmlToObject() {
+    XStream xstream = new XStream(new WstxDriver());
+    FeatureCollectionConverterWfs11 fcConverter = new FeatureCollectionConverterWfs11();
+    Map<String, FeatureConverter> fcMap = new HashMap<>();
+
+    GenericFeatureConverter converter = new GenericFeatureConverter();
+
+    fcMap.put("video_data_set", converter);
+    fcConverter.setFeatureConverterMap(fcMap);
+
+    xstream.registerConverter(fcConverter);
+
+    converter.setMetacardType(buildMetacardType());
+    xstream.registerConverter(converter);
+    xstream.registerConverter(new GmlGeometryConverter());
+    xstream.alias("FeatureCollection", WfsFeatureCollection.class);
+    InputStream is =
+        GenericFeatureConverterTest.class.getResourceAsStream(
+            "/video_data_set_collection_with_featureMembers.xml");
+
+    WfsFeatureCollection wfc = (WfsFeatureCollection) xstream.fromXML(is);
+    assertThat(wfc.getFeatureMembers(), hasSize(4));
+    Metacard mc = wfc.getFeatureMembers().get(0);
+    assertThat(mc.getId(), is("video_data_set.1"));
+  }
+
+  @Test(expected = IllegalArgumentException.class)
+  public void testUnmarshalNoMetacardTypeRegisteredInConverter() throws Throwable {
+    XStream xstream = new XStream(new WstxDriver());
+    xstream.registerConverter(new GenericFeatureConverter());
+    xstream.registerConverter(new GmlGeometryConverter());
+    xstream.alias(FEATURE_TYPE, Metacard.class);
+    InputStream is = GenericFeatureConverterTest.class.getResourceAsStream("/video_data_set_1.xml");
+    try {
+      xstream.fromXML(is);
+    } catch (Exception e) {
+      throw e.getCause();
```

Comment:
What's the purpose of throwing here?

Reply:
It looks like the original author wanted to check the root exception, not the top-level exception. But I don't have a better idea than that.